### PR TITLE
Fix #1022: will return errormessage when unknown command is given.

### LINKED ISSUE #1022: IXI NullPointerException on unknown commands
### Bug description
when an existing IXI package is called with an unknown command an NPE is thrown:
```
curl http://localhost:14265   -X POST   -H 'Content-Type: application/json'   -H 'X-IOTA-API-Version: 1'   -d '{"command": "Snapshot.alon"}'
```

```
09/27 19:08:00.027 [XNIO-1 task-8] ERROR com.iota.iri.service.API - API Exception: null
java.lang.NullPointerException: null
	at com.iota.iri.IXI.processCommand(IXI.java:192) ~[classes/:na]
	at com.iota.iri.service.API.process(API.java:329) [classes/:na]
	at com.iota.iri.service.API.processRequest(API.java:184) [classes/:na]
	at com.iota.iri.service.API.access$000(API.java:61) [classes/:na]
	at com.iota.iri.service.API$1.handleRequest(API.java:143) [classes/:na]
	at io.undertow.server.Connectors.executeRootHandler(Connectors.java:208) [undertow-core-1.4.6.Final.jar:1.4.6.Final]
	at io.undertow.server.HttpServerExchange$1.run(HttpServerExchange.java:809) [undertow-core-1.4.6.Final.jar:1.4.6.Final]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [na:1.8.0_181]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [na:1.8.0_181]
	at java.lang.Thread.run(Thread.java:748) [na:1.8.0_181]
```

### Steps To Reproduce
1. load `snapshot` IXI module
2. run the `curl` command above
3. see IRI log

### Expected behaviour
Similar response to a missing package, or any clear response, also on the client side.

### Actual behaviour
NPE on host & nothing on client

### PR DESCRIPTION
**Pull request by legacycode**
# Description

This PR fixes the NPE on unknown commands. I also added a parameter verification check and return an error message as well.

Fixes #1022 

## Type of change

- Bug fix (a non-breaking change which fixes an issue)

# How Has This Been Tested?

- I added unit tests and run mvn clean install

# Checklist:

- [x] My code follows the style guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/main/java/com/iota/iri/IXI.java [hunk 1]
```diff
@@ -202,7 +210,7 @@ private void loadModule(Path modulePath) {
             log.info("No package.json found in " + modulePath);
             return;
         }
-        final Map packageJson;
+        Map packageJson;
```
Comment (legacycode) [inline src/main/java/com/iota/iri/IXI.java]: Removing the final keyword is one of your PMD convention.

File: src/test/java/com/iota/iri/IXITest.java [hunk 2]
```diff
@@ -1,79 +1,80 @@
...
-        // Allow IXI to load the file
-        Map<String, Object> request = new HashMap<>();
-        Thread.sleep(1000);
-        response = IXI.instance().processCommand("test.getParser", request);
-
-        assertFalse(response instanceof ErrorResponse);
-        assertTrue(response instanceof IXIResponse);
+    public void processCommandError() {
+        AbstractResponse response = ixi.processCommand("testCommand.testSuffix", null);
+        assertNotNull("Response should not be null", response);
```
Comment (legacycode) [inline src/test/java/com/iota/iri/IXITest.java]: I am not sure if it makes sense to force comments on method level and assertion level. The tests are very small and sometimes documentation is not necessary.

Comment (GalRogozinski) [reply src/test/java/com/iota/iri/IXITest.java]: I will turn this off for tests

File: src/test/java/com/iota/iri/IXITest.java [hunk 3]
```diff
@@ -1,79 +1,80 @@
...
-        ixiResponse = ((IXIResponse) response);
-        assertNotNull(ixiResponse.getResponse());
-        */
+    /**
+     * If null is given as a command, expect a parameter check error message.
+     */
+    @Test
+    public void processCommandNull() {
+        AbstractResponse response = ixi.processCommand(null, null);
+        assertNotNull("Response should not be null", response);
```
Comment (GalRogozinski) [inline src/test/java/com/iota/iri/IXITest.java]: Can you add  `assertThat("Wrong type of response", response, CoreMatchers.instanceOf(ErrorResponse.class));`

Do it on the other tests as well

Comment (legacycode) [reply src/test/java/com/iota/iri/IXITest.java]: yes, checking for the correct instance type is better... done.

File: src/test/java/com/iota/iri/IXITest.java [hunk 4]
```diff
@@ -1,79 +1,82 @@
...
 import org.junit.BeforeClass;
 import org.junit.Test;
 import org.junit.rules.TemporaryFolder;
 
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Unit tests for {@link IXI}
+ */
```
Comment (GalRogozinski) [inline src/test/java/com/iota/iri/IXITest.java]: Just that you know, I don't mean to enforce javadocs on unit tests.
I have nothing against it  of course :-)

**Pull request merged**